[Bug]: error while building plugin

### Is there an existing issue already for this bug?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [x] I have read the troubleshooting guide and I think this is a new bug.

### What happened?

Recent versions of `github.com/google/cel-go` introduced a change incompatible with k8s.io/apiserver@v0.32.3, resulting in these compilation errors:

```
../../../go/pkg/mod/k8s.io/apiserver@v0.32.3/pkg/cel/environment/base.go:176:19: cannot use ext.TwoVarComprehensions (value of type func(options ...ext.TwoVarComprehensionsOption) "github.com/google/cel-go/cel".EnvOption) as func() "github.com/google/cel-go/cel".EnvOption value in argument to UnversionedLib
```

This can be fixed with:

```
replace github.com/google/cel-go => github.com/google/cel-go v0.22.1
```

See: https://github.com/kubernetes/apiserver/issues/116

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	github.com/cloudnative-pg/cnpg-i v0.2.0
 	github.com/cloudnative-pg/machinery v0.2.0
 	github.com/go-logr/logr v1.4.2
-	github.com/grpc-ecosystem/go-grpc-middleware/v2 v2.3.1
+	github.com/grpc-ecosystem/go-grpc-middleware/v2 v2.2.0
 	github.com/onsi/ginkgo/v2 v2.23.4
 	github.com/onsi/gomega v1.37.0
 	github.com/snorwin/jsonpatch v1.5.0

